\documentclass[12pt]{amsart}
\usepackage{natbib}

\def\N{\mathbb N}
\def\R{\mathbb R}
\def\P{\mathbb P}
\def\N{\mathbb N}
\def\S{\mathcal S}
\def\n{\mathbf n}
\def\x{\mathbf x}
\def\s{\mathbf s}
\def\y{\mathbf y}
\def\o{\%*\%}

\textwidth 7in
\oddsidemargin -0.25in
\evensidemargin -0.25in
\begin{document}
\bibliographystyle{plainnat}


\title[Extinction in Individual-based IPMs]{Individual-based Integral Projection Models: The role of size-structure on
  extinction risk and establishment success}
\author{Sebastian J. Schreiber and Noam Ross}
\address{Department of Evolution and Ecology, One Shields Avenue, University of California, Davis, California 95616}
\email{sschreiber@ucdavis.edu}
\maketitle{}

\section*{Introduction}
Computations of extinction probabilities or likelihoods of establishment success lie on opposing sides of a theoretician's coin and have been used to address theoretical and practical issues in conservation biology, restoration ecology, biological invasions and population genetics. Risks of extinction or establishment failure stem from populations consisting of a finite number of individuals, each of which faces a non-zero risk of mortality on any given day. These extinction risks are shaped, in part, by the size and composition of a population whose individuals may differ in age, size, geographical location, or other important characteristics influencing demography. When population structure is finite-dimensional (i.e. a finite number of age classes, stages, geographical locations), multi-type branching processes can model these extinction risks and, thereby, serve as the stochastic, finite-population counterpart of matrix models~\citep{harris-63,athreya-ney-04,caswell-01,haccou-etal-05}. These stochastic models have been used successfully to address a diversity of questions concerning fixation probabilities of beneficial alleles~\citep{patwa-wahl-08}, evolutionary emergence of pathogens~\citep{antia-etal-03,ptrb-13}, extinction risk of small populations~\citep{boyce-92,gosselin-lebreton-00,fujiwara-caswell-01,erickson-etal-15}, and establishment success in heterogeneous environments~\citep{haccou-iwasa-96,haccou-vatunin-03,amnat-09b}.

To parameterize matrix models or mutli-type branching processes, individuals must be discretely categorized into a finite number of types. However, when collecting demographic data, researchers commonly measure continuous traits (e.g. mass, length, geographical location) about individuals, as well as discrete traits. Integral projection models (IPMs) allow one to account for this continuous population structure~\citep{easterling-etal-00}.These IPMs can be viewed as infinite-dimensional matrix models and can be numerically approximated by finite-dimensional matrix models. Consequently, many of the standard demographic concepts and methods for matrix models (e.g. stable state distributions, reproductive values, life table response experiments, sensitivity analysis) exist for IPMs~\citep{easterling-etal-00,rees-ellner-09,coulson-12,metcalf-etal-13,rees-etal-14,merow-etal-14}.

Here, we describe individual-based counterparts of IPMs using continuous-state branching processes~\citep{harris-63}. For these finite population, stochastic models, we present an analytical method for computing extinction probabilities. As these methods are easily implemented numerically, they circumvent the need to use individual-based simulations and allow one to efficiently study how extinction probabilities or establishment failure depend on continuous as well as discrete population structure. We illustrate the application of these methods with an individual-based IPM of the endangered alpine plant \emph{Dracocephalum ausriacum} using a mean-field IPM developed by \citet{nicole-etal-11}.


\section*{The General Models and Methods}


\subsection*{The Individual Based IPM}
We consider an individual-based model where the set of all possible individual states (e.g. age, size, geographical location, etc.) lies in a compact metric space $X$. For the standard size-structured IPM, $X=[a,b]$ corresponds to the range of sizes measured in the field where $a$ is the minimal size and $b$ is the maximal size. For models with a mixture of age and size structure, $X$ could be given by $\{1,\dots,T\}\times [a,b]$ where $T$ corresponds to the maximal age of an individual.

Following \citet{harris-63},  we consider finite populations in which the state of the population at any point of the time is characterized by the different states $(x_1,x_2,\dots, x_k)$  of individuals within the population and the number of individuals in each state $(n_1,n_2,\dots, n_k)$. Specifically, if there $n_1$ individuals in state $x_1$, $n_2$ individuals in state $x_2$,..., $n_k$ individuals in state $x_k$, then the state of the population is given by $\s=(n_1,n_2,\dots, n_k; x_1,x_2,\dots, x_k)$. The set of all possible population states is
\[
\S= \{(\n,\x)=(n_1,\dots, n_k, x_1\dots , x_k): k,n_i\in \N, x_i \in X\} \cup \{0\}
\]
where $\N=\{1,2,3,\dots\}$ denotes the natural numbers and $0$ is the extinction state corresponding to no individuals in the population.

Let  $\s_t=( \n, \x)\in \S$ be the population state at time $t$. The dynamics of $\s_t$ are determined by a set of probabilistic rules that determine the contribution of each individual in the population to the population in next time step $t+1$. These ``contributions'' may correspond to an individual surviving and changing state (e.g. growing in size, getting older, dispersing to another geographical location), or having offspring.  Consistent with standard branching process theory, each individual in the population updates independent of all other individuals.

The update rule for an individual in state $x$ is given by a probability measure $m(x,d\s)$ on the state space $\S$. Specially, the probability an individual in state $x$ contributes $\s$ individuals to the population in the next time step where $\s$ lies in a subset $A\subset \S$ equals
\[
\P[\s_1\in A| \s_0 =(1,x)]=\int_A m(x,d\s)
\]
where the left hand side reads ``the probability the population state lies $A$ at time $1$ after initially having only one individual in state $x$ at time $0$." If the population state is currently $\s_t=(n_1,\dots,n_k,x_1,\dots,x_k)$, then the state $\s_{t+1}$ is determined as follows:
\begin{enumerate}
\item for each of the $n_1$ individuals in state $x_1$, randomly and independently choose the number of replacement individuals from distribution $m(x_1,d\,\s)$,
\item repeat step (1) for the states $x_2,\dots, x_k$, and
\item determine the new population state $\s_{t+1}$ by identifying the states of all individuals and counting the total number of individuals in each of these states.
\end{enumerate}
This iterative algorithm can be used to create individual based simulations of the individual based IPM. As with any branching process, stochastic realizations of this process, with probability one, either go to extinction in finite time, or the population abundance grows without bound. This latter event is typically interpreted as a population becoming established and persisting.

\subsection*{Probability generating functionals and extinction probabilities}

We can characterize the probabilistic state of the system using probability generating functionals $\Psi$ (pgfs). Unlike moment generating functionals as used by \citet{harris-63}, the pgfs allow us to directly compute how extinction probabilities change in time as well as compute asymptotic extinction probabilities.

To define the pgf $\Psi$, we introduce the following notation: given a continuous function $h:X\to\R$ and $\s\in \S$, let
\[
h^\s=
\left\{
\begin{array}{cc}
\prod_{i=1}^k h(x_i)^{n_i}&\mbox{ if }\s=(n_1,\dots,n_k,x_1,\dots,x_k)\\
1 &\mbox{ if }\s=0.
\end{array}
\right.
\]
The probability generation functional is defined by
\begin{equation}\label{*}
\Psi(h)(x)=\int h^\s \,m(x,d\s).
\end{equation}
$\Psi(h)(x)$ corresponds to the expected value of $h^\s$ due to the contributions of individuals in the next time step from an individual in state $x$. This expected value requires integrating over all possible populations states in the next time step.

The utility of $\Psi$ for computing extinction probabilities follows from two facts. First, the definition of $\Psi$ implies that if $h_0$ is the zero function (i.e. $h_0(x)=0$ for all $x$), then
\begin{equation}\label{star}
\Psi(h_0)(x)=\int 1 \,m(d\s,x)= \P[\s_1=0|\s_0=(1,x)]
\end{equation}
is the probability the population goes extinct in one time step given that initially it consisted of one individual is state $x$. For the second fact, define
\[
\Psi^t(h)=\underbrace{\Psi(\Psi(\dots \Psi}_{\mbox{$t$ times}}(h)\dots))
\]
to be the $t$-fold composition of $\Psi$ with itself. We claim that
\begin{equation}\label{**}
\Psi^t(h_0)(x)=\P[\s_t=0|\s_0=(1,x)]=:E_t(x)
\end{equation}
is the probability of extinction by time $t$ given that the population initially consisted of one individual in state $x$. To verify this claim, we argue by induction. Equation~\ref{star} implies that equation \eqref{**} holds for $t=1$. Now suppose that equation \eqref{**} holds for $t$, we will show it holds for $t+1$.  On the event that $\s_1=(n_1,\dots,n_k,x_1,\dots,x_k)$ is the population state at time $1$, extinction occurs by time $t+1$ only if each of the lineages of the $n_1+\dots +n_k$ individuals go extinct in the next $t$ time steps. As the fates of these lineages are independent of one another, it follows that
\[
\begin{aligned}
\P[s_{t+1}=0|\s_0=(1,x),\s_1=(n_1,\dots,n_k,x_1,\dots,x_k)]&= \prod_{i=1}^k \P[\s_t=0|\s_0=(1,x_i)]^{n_i}\\
&=\prod_{i=1}^k\left(\Psi^t(h_0)(x_i)\right)^{n_i}=\Psi^t(h_0)^{\s_1}
\end{aligned}
\]
where the second equality follows from our inductive hypothesis. By the law of total probability
\[
E_t(x)=\P[s_{t+1}=0|\s_0=(1,x)]=\int [\Psi^t(h_0)]^\s \,m(d\s,x)
\]
which, by definition, equals $\Psi^{t+1}(h_0)(x)$ as claimed.

Equation \eqref{**} can be used to compute extinction probabilities iteratively. Furthermore, as individuals update independent of one another, the probability of the population going extinct by time $t$ for any initial condition $\s=(n_1,\dots,n_k,x_1,\dots,x_k)$ equals
\begin{equation}\label{starstar}
\P[\s_t=0|\s_0=\s]=E_t^\s.
\end{equation}
These analytic expressions allow us to efficiently compute extinction probabilities by constructing a numerical approximation of the pgf $\Psi$ and iterating it with an initial condition of a zero vector which corresponds to the numerical approximation of the zero function.

As $E_0(x)\le E_1(x),\le E_2(x)\dots$ for any $x\in X$ and $E_t(x)\le 1$ for all $t$, there is a well defined limit corresponding to the probability of eventual extinction:
\[
E_\infty(x):=\lim_{t\to\infty} E_t(x).
\]
Using moment generating functionals with suitable technical hypotheses, \citet{harris-63} showed that $E_\infty(x)<1$ for all $x$ if the dominant eigenvalue of the mean-field IPM is greater than one, and $E_\infty(x)=1$ for all $x$ otherwise.

\section*{An Illustration with an Endangered Plant Species}

We illustrate these general methods using an individual-based IPM for the endangered alpine plant, the Austrian Dragonhead (\emph{D. austriacum L.,Lamiaceae}). \emph{D. ausriacum} is a long-lived perennial plant that is listed as Vulnerable by the World Conservation Union.  This illustrative model is an individual-based counterpart of an IPM developed by \citet{nicole-etal-11} for this plant, based on a subset of their data.

In the model, the state $x$ of an individual is the size which corresponds to the logarithm of the product of plant height and number of stems. If $N_t(x)$ denotes the density of individuals of size $x$ in year $t$, then \citet{nicole-etal-11} used an IPM of the form
\[
N_{t+1}(x)=\int_a^b \left[ s(y)G(y,x)+e(x)p(y)f(y)N_t(y)\right]\,dy
\]
where $s(y)$ is the probability of surviving to the next year for individuals of size $y$, $G(y,x)dy$ is the infinitesimal probability that a surviving individual of size $y$ is size $x$ in the next year, $p(y)$ is the probability that an individual of size $y$ flowers, $f(y)$ is the mean number of offspring produced by an individual of size $y$, and $e(x)$ is the infinitesimal probability of an offspring is of size $x$ at the time of the annual census.

Following \citet{nicole-etal-11}, we used generalized linear models (GLMs) to select functional forms for each of the demographic kernels. Computations were performed using the GLM package for R \citep{R-15}. Specifically, we used logistic regression (i.e. a GLM with the binomial family) for determining $s(y)$ and $p(y)$, a GLM with the Poisson family for modelling $f(y)$, and modeled $e(y)$ using a normal distribution whose mean and variance where determined by the empirical distribution of germinants. For the growth kernel, linear regression determined the expected size of an individual in the next year and the actual size was assumed to be normally distributed about this mean. The variance of this normal distribution, for simplicity, was assumed to be independent of the current size of an individual. Figure 1 shows the data and fits for $s(y)$, $G(y,x)$, $p(x)$ and $f(x)$.



<<boring-plots, fig.width=7, fig.height=5,echo=FALSE,fig.cap="The demographic kernels for Austrian Dragonhead with the corresponding data from Nicole et al. (2011).">>=
rm (list = ls ())
load("IPMs.Rdata")
g=function(y,x)Growth(y,x)
s=function(x)Survival(x)
f=function(x)fecundity(x)*establishment.prob
pf=function(x)1
e=function(y)dnorm(y,mean=recruit.size.mean,sd=recruit.size.sd)
par(mfrow=c(2,2),mar=c(4.5,4.5,1,1),cex.lab=1.5,cex.axis=1.5)
plot(data$size,jitter(data$surv,factor=0.1),pch=21,bg=rgb(1,0,0,0.5),xlab="size x",ylab="survival probability")
lines(xs,Survival(xs),lwd=4,col="blue")
image(xs,xs,t(outer(xs,xs,Growth)),xlab=expression(size[t]),ylab=expression(size[t+1]))
points(data$size,data$sizeNext,pch=21,bg=rgb(0,0,1,0.5))
plot(data$size,jitter(data$fec.flower,factor=0.1),pch=21,bg=rgb(1,0,0,0.5),xlab="size",ylab="flowering")
lines(xs,flowering(xs),lwd=4,col="blue")
plot(size.temp,seed.temp,pch=21,bg=rgb(1,0,0,0.5),bty="n",xlab="size",ylab="fecundity")
lines(xs,fecundity(xs),lwd=2)
@

The kernels $s(x)$, $G(y,x)$, $p(x)$ and $e(x)$ provide us with all the information that the individual-based IPMs requires for probabilistic updating individuals for survival, growth, flowering or not, and size of germinating individuals at first census. The fecundity kernel $f(y)$, however, only specifies the mean number of offspring produced, but for an individual-based IPM, we need the distribution of the number of offspring produced by an individual. Fortunately, this information is built into the structure of the GLMs, based on the assumptions in our model choice. For example, the fecundity data was modeled using a Poisson family for the GLM.  In which case, the mean number $f(x)$ of offspring also specifies the distribution. More generally, one might use multi-parameter distributions such as a zero inflated Poisson or a negative binomial in which case parameters in addition to the mean are needed to specify the distribution of offspring number.

\subsection*{Deriving the pgf $\Psi$}

To define $\Psi$, we observe that the contributions of an individual of size $x$ to the population in the next time step involves the sum of two independent random variables:  the contribution due to survival and growth and the contribution due to reproduction. We will identify two pgfs, $\Psi_g$ and $\Psi_f$, for each of these processes separately. Then, we make use of a fundamental property of pgfs:\vskip 0.1in
\fbox{\parbox{\textwidth}{\textbf{Fundamental Property 1}:
The pgf for a sum of independent random variables is the product of the pgfs of these random variables.}}
\vskip 0.1in
to get that
\[
\Psi=\Psi_g \times \Psi_f.
\]
When the random variables are identically, distributed, we have the following generalization of this fundamental property of pgfs:
\vskip 0.1in
\fbox{\parbox{\textwidth}{\textbf{Fundamental Property 2}: The pgf for a sum of $N$ independent, identically distributed random variables $X_i$ is the composition of the pgf for $N$ with the pgf for the $X_i$.}}\vskip 0.1in


For the pgf $\Psi_g$ for survival and growth, $\Psi_g(h)(x)$ corresponds to integrating $h^\s$ over all possible contributions $\s$ from an individual of size $x$ surviving and growing. These contributions are of two types: $\s=0$ when the individual dies,  and $\s=(1,y)$ when the individual survives and grows to size $y$. The first event occurs with probability $1-s(x)$ and the infinitesimal probability of the second event is $s(x)G(y,x)dy$. As $h^\s=1$ when $\s=0$ and $h^\s=h(y)$ when $\s=(1,y)$, integrating over all possible contributions due to survival and growth yields
\[
\Psi_{g}(h)(x)=(1-s(x))\times 1 + s(x)\int h(y)G(y,x)dy.
\]

For the pgf $\Psi_f$ corresponding to contributions due to fecundity, $\Psi_f(h)(x)$ is given by integrating $h^\s$ over all possible states $\s$ corresponding to the offspring produced by an individual of size $x$.  To write this down, we make two applications of  \textbf{Fundamental Property 2} of pgfs. We begin by conditioning on the event that an individual of size $x$ flowers. On this event, the individual produces a Poisson number $N$ of offspring with mean $f(x)$. The size of each of these offspring is drawn interdependently from the common offspring distribution $e(y)dy$. Hence, the contribution of a flowering individual of size $x$ is the sum of $N$ independent random variables with distribution $e(y)dy$. By \textbf{Fundamental Property 2} of pgfs, we need to take the composition of the pgf for $N$, namely $\phi(x,\cdot)$, with the pgf for a single offspring, namely
\[
\Psi_e(h)(x)=\int h(y)e(y)dy.
\]
Thus, we get the pgf associated with a flowering individual is
\[
\Psi_{flower}(h)(x)=\phi(x,\int h(y)e(y)dy)
\]
To get the pgf for flowering and non-flowering contributions to fecundity, we observe that fecundity contributions of an individual of size $x$ is given by the sum of a Bernoulli number of flowering individuals where the probability of success is $p(x)$. By the \textbf{Fundamental Property 2}, we need to compose the pgf of a Bernoulli, which is $\theta(s)=1-p(x)+p(x)s$ where $s$ is the dummy variable, with the pgf of a flowering individual. This composition yields
\[
\Psi_f (h)(x)=\theta(\Psi_{flower}(h)(x))=1-p(x)+p(x)\phi(x,\int h(y)e(y)dy).
\]


Putting together the two pieces, we get the desired pgf $\Psi$:
\[
\Psi(h)(x)=\left((1-s(x))\times 1 + s(x)\int h(y)G(y,x)dy\right)\phi(x,\int h(y)e(y)dy).
\]


\subsection*{Numerically Implementing and Using the pgf $\Psi$}

To approximate $\Psi$ numerically, we discretize a larger size interval $[0,2b]$ using $n$ equal sized intervals of width $\Delta x=(2b)/n$. We used the larger size interval of $[0,2b]$ to minimize the effects of eviction (see below). To use the midpoint rule to approximate the integrals, we created a vector $\vec x$ corresponding to the midpoints of these intervals. Using this vector we discretized the survival function as a vector $\vec s=s(\vec s)$, the growth kernel as a matrix using the outer product $G$ of the growth kernel $g$ with $\vec x$, the probability of flowering function as a vector $\vec p=p(\vec x)$, the fecundity function as a vector $\vec f=f(\vec x)$, and the offspring size distribution as a vector $\vec e=e(\vec x)$.

For the previously described methods to work, it is critical that column sums for $g.mat$ and the sum of $e.vec$ equal one. For most IPMs, this will not occur automatically due to individuals being evicted from the size interval $[0,2b]$. There are a variety of ways to handle this issue~\citep{williams-etal-12}. As the  offspring size vectors nearly summed to one, we simply re-normalized it so that it summed to one. For the growth matrix, we treated eviction as mortality which is consistent with most eviction events corresponding to growing to a negative size. To do this, we took one minus the column sums and subtracted them from the survival vector and then normalized the column sums so that they added to one. When taking the product of survival and growth, the mean-field IPM is unaffected by this change.

Using these discretized demographic components and the pgf $\phi(x,s)=\exp(-f(x)(s-1))$ for fecundity, we get the discretized pgf $\Psi$ is given by
\[
\Psi_{discrete}(\vec h)=(1-\vec s+\vec s\circ (G^T\, \vec h))\circ(1-\vec p+\vec p\circ \phi(\vec x,\vec e^T\,\vec h))
\]
where $\vec h$ corresponds to a discretized function i.e. a vector of length $n$, $^T$ denotes the transpose of a matrix or vector,  and $\circ$ denotes element by element multiplication.


<<main.IBIPM.code,echo=FALSE>>=
n=100 # number of bins
xs=seq(alpha,2*beta,length=n)
dx=xs[2]-xs[1]
s.vec=s(xs)*0.95
e.vec=e(xs)*dx
flowering.vec=as.numeric(flowering(xs))
e.vec=e.vec/sum(e.vec)
g.mat=outer(xs,xs,g)*dx
q.evict=1-colSums(g.mat)
g.mat=g.mat%*%diag(1/(1-q.evict))
# Note: most of the eviction is from negative sized individuals
s.vec=(1-q.evict)*s.vec
phi=function(x,s)exp(f(x)*(s-1))
Psi=function(h){
h.new=(s.vec*(t(g.mat)%*%h)+1-s.vec)*(1-flowering.vec+flowering.vec*phi(xs,e.vec%*%h))
return(h.new)
}
Psi.iterate=function(T){
hs=matrix(0,T+1,n)
for(t in 1:T){
hs[t+1,]=Psi(hs[t,])
}
return(hs)
}
# function for estimating asymptotic extinction probability
AE=function(tol=0.0001){
h=rep(0,n)
diff=1
while(diff>tol){
h.old=h
h=Psi(h)
diff=sum(abs(h.old-h))
}
return(h)
}

#Inputs: N vector of length $n$ which specifies the initial number of individuals in each size bin, and T the length of the time to look things over.
#Output:a vector of length T+1 corresponding to the probabilities of extinction by time t or sooner
PVA=function(N,T){
hs=Psi.iterate(T)
output=numeric(T+1)
for(t in 1:(T+1)){
output[t]=prod(hs[t,]^N)
}
return(output)
}
@


Iterating $\Psi$ yields how extinction probabilities $E_t(x)$ vary with time, as illustrated in Figure 2. Intuitively, this figure illustrates that the probability of extinction decreases with the size of individual initially founding the population, and that extinction probabilities increase over time. Furthermore, the $E_t(x)$ over time are approaching limiting extinction probabilities $E_{\infty}(x)$ which are less than one. This stems from the fact that the dominant eigenvalue of the mean-field IPM is greater than one. To estimate this asymptotic probability of extinction we  iterated $\Psi$ until a tolerance condition is met e.g. $|E_{t+1}(x)-E_t(x)|\le 0.0001$ for all $x$ (see thick gray curve in Figure 2).



<<IBIPM-plots2, fig.width=6, fig.height=4,echo=FALSE,fig.cap="Extinction probabilities for a population consisting of a single individual of size $x$. Different curves correspond to extinction in different number of years.">>=
T=100
out=Psi.iterate(T)
times=c(1,5,10,25,30,40)+1
par(cex.lab=1.25,cex.axis=1.25)
matplot(xs,t(out[times,]),xlim=c(alpha,beta),type="l",lty=1,xlab="size",ylab="extinction probabilities",bty="n",lwd=2)
legend("topright",c(as.character(times-1),expression(Inf)),col=c(1:length(times),rgb(0,0,0,0.5)),lty=1,lwd=c(rep(2,length(times)),5),bty="n")
# adding the asymptotic extinction probability
h=AE()
lines(xs,h,type="l",lwd=5,xlim=c(alpha,beta),col=rgb(0,0,0,0.5))
#legend("topleft","A",bty="n",cex=1.25)
@

To scale things up to an entire population initially in state $\s_0$, recall that the extinction probability at time $t$ is $E_t^{\s_0}$. Figure 3 illustrates how the extinction probabilities over time vary for a population with initially $100$ individuals of the smallest size $\alpha$ (i.e. $\s_0=(\alpha,100)$) versus a population with $5$ or $6$ individuals of the  size $\beta/2$ (i.e $\s_0=(\beta/2,5)$ or $(\beta/2,6)$. Figure 3 suggests that, from the extinction risk perspective, about $6$ larger individuals are equivalent to $100$ of the smallest individuals. These types of comparisons may be particularly useful when trying to assess whether planting small or larger individuals are more effective for establishment success.

<<IBIPM-plots3, fig.width=6, fig.height=4,echo=FALSE,fig.cap="Extinction probabilities as a function of time for populations of different abundances and size distributions.">>=
T=75
N=rep(0,n)
N[1]=100
out1=PVA(N,T)
N=rep(0,n)
N[50]=6
out2=PVA(N,T)
N=rep(0,n)
N[50]=5
out3=PVA(N,T)
out=cbind(out1,out2,out3)
par(cex.lab=1.25,cex.axis=1.25)
matplot(0:T,out,typ="l",lwd=3,bty="n",xlab="time t",ylab="probability of extinction by year t")
legend("topleft",c("100 small","6 large","5 large"),lty=1:3,col=1:3,lwd=3,bty="n")
legend("topright","B",bty="n",cex=1.25)
@

We also used the data to see how extinction risk over different time frames depends on the size and composition of the population. Specifically, for different population abundances $N$, we randomly sampled $N$ individuals from the data and computed extinction risk of this sampled population over $5$ and $50$ year periods (Figure 4). For each population abundance $N$, we considered $500$ random samples of size $N$. Figure 4A illustrates that $\log$ extinction risk decreases, on average, linearly with the sample size and is, on average, greater for the $50$ year period than the $5$ year period. However, for smaller sample sizes, there is overlap in the distributions of extinction times for the $5$ and $50$ year time frames. This overlap stems from  samples of mostly large individuals being more likely to persist at least $50$ years than samples of mostly small individuals persisting at least $5$ years. Consistent with this explanation, Figure 4B illustrates that mean size of an individual within the population has a strong negative correlation with (log) extinction probability, and the slope of this correlation is steeper over shorter time frames than longer time frames.




<<IBIPM-plots4,cache=TRUE, fig.width=7, fig.height=4,echo=FALSE,fig.cap="Extinction probabilities for populations of different abundance sampled for the size distribution data. For each population abundance $N$, extinction probablities by year $5$ and year $50$ were calculated for $500$ samples of size $N$. In A, extinction probability is plotted as a box plot for different population abundances and different time frames. In B, extinction probability is plotted against the mean size of an individual for a population abundance $N=25$.">>=
sizes=data$size[1:400] # empirical data
T=51
T2=5
sample.sizes=5*(1:5)
no.samples=500
stuff=c()
mean.size=c()
stuff2=c()
temp=numeric(no.samples)
temp2=temp
temp3=temp
for(j in 1:length(sample.sizes)){
for(i in 1:no.samples){
sampling=sample(sizes,size=sample.sizes[j])
out2=hist(sampling,breaks=xs,plot=FALSE)
N=c(0,out2$counts)
out=PVA(N,T)
temp[i]=out[T+1]
temp2[i]=out[T2+1]
temp3[i]=mean(sampling)
}
stuff=cbind(stuff,temp)
stuff2=cbind(stuff2,temp2)
mean.size=cbind(mean.size,temp3)
}
par(cex.lab=1.25,cex.axis=1.25,mfrow=c(1,2),mar=c(4.5,4.5,1,1))
boxplot(log10(stuff),notch=FALSE,names=sample.sizes,col="lightgray",range=0,xlab="population abundance",ylab=expression(paste(log[10]," extinction probability")),ylim=c(min(log10(stuff2)),max(log10(stuff))))
boxplot(log10(stuff2),notch=FALSE,names=sample.sizes,col="pink",range=0,add=TRUE)
legend("bottomleft",c("in 5 years","in 50 years"),pch=22,pt.bg=c("pink","white"),cex=1.25,bty="n")
legend("topright","A",bty="n",cex=1.25)
par(mar=c(4.5,2.25,1,2.25))
i=5;plot(mean.size[,i],log10(stuff[,i]),ylim=c(min(log10(stuff2[,i])),max(log10(stuff[,i]))),xlab="mean size of an individual",ylab="",pch=20,col="lightgray");points(mean.size[,i],log10(stuff2[,i]),pch=20,col="pink")
legend("bottomleft",c("in 5 years","in 50 years"),pch=c(20,20),col=c("pink","lightgray"),cex=1.25,bty="n")
legend("topright","B",bty="n",cex=1.25)
legend("center","population abundance\n            = 25",bty="n",cex=1.25)
@

\section*{Recommendations, Extensions, and Future Challenges}

To implement the methods presented here, there are two main steps. First, one needs to identify the main demographic processes of the population, the order in which these processes occur relative to the censuses used for data collection, and develop the statistical models for the each of the demographic processes. \citet{rees-etal-14} and \citet{merow-etal-14} provide excellent reviews on the philosophical and methodological issues associated with this step for mean-field IPMs. Typically, whenever a study has sufficient data for constructing the mean-field IPM, there is no need to collect any additional data to build the individual-based IPM.  Rather, unlike the mean-field IPMs, the individual-based IPMs make use of the complete distributional information associated with fecundity. As with other areas of stochastic demography, the shapes, not just the means, of these distributions may have significant effects on the likelihood of extinction or establishment success. Hence, it is best to examine several options (e.g. Poisson versus negative Binomial versus zero-inflated distributions) to identify which distribution does a better job of describing the fecundity data. As in all areas of modeling, if there is significant uncertainty about the ``best'' choice of the model, one should perform the analyses with each of the alternative fecundity distributions to identify the sensitivity of predictions of extinction risk to these alternatives.

The second step, the focus of this paper, involves constructing the probability generating functional $\Psi$. For the uninitiated, this step may be intimidating. However, there are a three basic principles that simplify this construction. First, while this pgf $\Psi$ takes functions to functions, one should focus on writing down $\Psi(h)(x)$, which involves understanding the contributions of a single individual of size $x$ to the next census. Second, one can often can break up these contributions into a sum of independent contributions, find the pgfs associated with these simpler contributions, and then use the two fundamental properties of pgfs to ``stitch" together $\Psi$. Third, the distributions used to describe the number of offspring produced by an individual typically involve random variables (e.g. Poisson, negative binomial) for which their associated pgf is well-known. Finally, whenever in doubt, find a collaborator that you trust to put the pieces together correctly.


For the individual-based models considered here, we assumed the environment remains constant over time. However, IPMs have and continue to be used to study the effects of fluctuating environmental conditions on population demography and life history evolution~\citep{childs-etal-04,dahlgren-ehrlen-11,rees-ellner-09}. The methods described here easily extended to fluctuating environments. Specifically, if $\Psi_t$ is the pgf of the individual-based IPM associated with year $t$, then the probability of extinction for a population initially in state $\s$ is
\[
\Psi_0(\Psi_1(\dots \Psi_t(h_0)))^\s
\]
where $h_0$, as before, is the zero function. Note that the composition here is in the reverse order of what one does when iterating the mean-field IPM model forward in time. While we know of no formal proof, in the case of a stationary environment with suitable technical assumptions, we conjecture the following limit theorem holds: the asymptotic extinction probabilities are strictly less than one if and only if the stochastic growth rate (aka dominant Lyapunov exponent) of the mean-field IPM is positive. For multi-type branching processes, this result was proven by \citet{tanny-81}.

Despite this relatively straightforward extension to temporally variable environments, many challenges remain. From the computational perspective, finding the efficient methods to deal with multi-dimensional states variables (e.g. size and location, or multi-dimensional traits) continues to be a challenge, as it is for mean-field IPMs. While the analytical methods presented here cover multi-dimensional state variables, their numerical implementation involves approximating mutli-dimensional integrals which can be computationally expensive. From the analytical perspective, accounting for temporal correlations in individual growth or reproductive rates (e.g. individuals that grew larger than expected in one year being more likely to grow larger than expected in the next year) or correlations among individuals are particularly important challenges as strong correlations likely have large effects on extinction risk.
Finally, and perhaps most importantly, the  manners in which size-structured demography may shape extinction risk for real-world population remains to be understood. One might hope that by applying these methods to the many data sets for which IPMs have been developed, as well as future data sets, might provide a computationally efficient means to explore how size-structure for populations differing in their evolutionary history and environmental context influences their vulnerability to extinction.

% Acknowledgements: Thanks to Jacob Moore for several excellent suggestions including the addition of Figure 4B.

\bibliography{seb}


\end{document}
